code/mob/living/critter/ai/shared.dm

```diff
@@ -131,14 +85,18 @@
 	var/max_path_dist = 50 //keeping this low by default, but you can override it - see /datum/aiTask/sequence/goalbased/rally for details
 	var/list/found_path = null
 	var/atom/move_target = null
-	var/move_through_space = FALSE
 
 // use the target from our holder
 /datum/aiTask/succeedable/move/proc/get_path()
 	if(!move_target)
 		fails++
 		return
-	src.found_path = get_path_to(holder.owner, move_target, src.max_path_dist, 0, null, !move_through_space)
+	if(length(holder.target_path) && holder.target_path[length(holder.target_path)] == move_target)
```

Comment:
this assumes `move_target` is always the turf and not the actual target, is it? (and if so it should probably be typed as a turf)

Reply:
it is, but yes, I will type it as turf for clarity.